code/game/gamemodes/gang/gang.dm

```diff
@@ -0,0 +1,177 @@
+			SSticker.mode_result = "win - gangs survived"
+			SSticker.news_report = GANG_OPERATING
+			return gangs_still_alive
+	SSticker.mode_result = "loss - gangs destroyed"
+	SSticker.news_report = GANG_DESTROYED
+	gangs_still_alive = 0
+
+/datum/game_mode/gang/check_finished()
+	if(fuckingdone)
+		return TRUE
+	else
+		return ..()
+
+/datum/game_mode/gang/process()
+	check_counter++
+	if(check_counter >= 5)
+		if (world.time > endtime && !fuckingdone)
+			fuckingdone = TRUE
+		check_counter = 0
+		SSticker.mode.check_win()
+
+		check_tagged_turfs()
+		check_gang_clothes()
+		check_rollin_with_crews()
+		// if we had an easy way to source what came from who, I'd add points for getting people addicted to drugs.
+
+
+/datum/game_mode/gang/proc/check_tagged_turfs()
+	for(var/obj/effect/decal/cleanable/crayon/gang/tag in GLOB.gang_tags)
+		if(tag.my_gang)
+			tag.my_gang.adjust_points(50)
+		CHECK_TICK
+
+/datum/game_mode/gang/proc/check_gang_clothes() // TODO: make this grab the sprite itself, average out what the primary color would be, then compare how close it is to the gang color so I don't have to manually fill shit out for 5 years for every gang type
+	for(var/mob/living/carbon/human/H in GLOB.alive_mob_list)
+		if(!H.mind)
+			continue
+		var/datum/antagonist/gang/is_gangster = H.mind.has_antag_datum(/datum/antagonist/gang)
+		for(var/clothing in list(H.head, H.wear_mask, H.wear_suit, H.w_uniform, H.back, H.gloves, H.shoes, H.belt, H.s_store, H.glasses, H.ears, H.wear_id))
+			if(is_gangster)
+				if(is_type_in_list(clothing, is_gangster.acceptable_clothes))
+					is_gangster.add_gang_points(10)
+			else
+				for(var/datum/team/gang/gang_clothes in gangs)
+					if(is_type_in_list(clothing, gang_clothes.acceptable_clothes))
+						gang_clothes.adjust_points(5)
+
+		CHECK_TICK
+
+/datum/game_mode/gang/proc/check_rollin_with_crews()
+	for(var/area/A in GLOB.sortedAreas)
+		var/list/gang_members = list()
+		for(var/mob/living/carbon/human/H in A)
+			if(H.stat)
+				continue
+			var/datum/antagonist/gang/is_gangster = H.mind.has_antag_datum(/datum/antagonist/gang)
+			if(is_gangster)
+				gang_members[is_gangster.my_gang]++
+			CHECK_TICK
+		if(gang_members.len)
+			for(var/datum/team/gang/gangsters in gang_members)
+				if(gang_members[gangsters] >= 4)
+					if(gang_members[gangsters] >= 8)
+						gangsters.adjust_points(5) // Discourage larger clumps, spread ur people out
+					else
+						gangsters.adjust_points(10)
+
+
+/datum/game_mode/gang/generate_report()
+	return "Something something grove street home at least until I fucked everything up idk nobody reads these reports."
+
+/datum/game_mode/gang/send_intercept(report = 0)
+	return
+
+/datum/game_mode/gang/special_report()
+	var/list/report = list()
+	var/highest_point_value = 0
+	var/highest_gang = "Leet Like Jeff K"
+	report += "<span class='header'>The gangs in the round were:</span>"
+	for(var/datum/team/gang/G in gangs)
+		report += "<span class='header'>[G.name]:</span>"
```

Comment:
I scrapped team report in favor of this setup so that I could make sure the formatting was in the correct order for gangs. I also need to re-do the formatting anyways and order it from highest/lowest points

Reply:
This locks this hard to the mode, should instead mark the participating teams as no show if you want to handle them yourself.

Reply:
A lot of the gang item stuff relies heavily on gang being the active gamemode, and this isn't intended to be a mulligan gamemode.

Reply:
It needs to be modularized enough for that to be a possibility though.

Reply:
Unless admin tools are able to call pre/post setup, that can't happen.

Reply:
that can't happen _in it's current form_ :^)